test/vfd_swmr_bigset_writer.c

```diff
@@ -942,33 +1483,110 @@ verify_extensible_dset(state_t *s, unsigned int which, mat_t *mat, unsigned *ste
 
             /* Down the right side, intersecting the bottom row. */
             base.col = last.col;
-            for (base.row = ofs; base.row <= last.row; base.row += s->chunk_dims[0]) {
-                verify_chunk(s, filespace, mat, which, base);
+            for (base.row = 0; base.row <= last.row; base.row += s->chunk_dims[0]) {
+                if (!repeat_verify_chunk(s, filespace, mat, which, base)) {
+                    fprintf(stderr, "chunk verification failed\n");
+                    TEST_ERROR;
+                }
             }
 
             /* Across the bottom, stopping before the last column to
              * avoid re-reading the bottom-right chunk.
              */
             base.row = last.row;
-            for (base.col = ofs; base.col < last.col; base.col += s->chunk_dims[1]) {
-                verify_chunk(s, filespace, mat, which, base);
+            for (base.col = 0; base.col < last.col; base.col += s->chunk_dims[1]) {
+                if (!repeat_verify_chunk(s, filespace, mat, which, base)) {
+                    fprintf(stderr, "chunk verification failed\n");
+                    TEST_ERROR;
+                }
             }
         }
         else {
-            verify_chunk(s, filespace, mat, which, last);
+            if (!repeat_verify_chunk(s, filespace, mat, which, last)) {
+                fprintf(stderr, "chunk verification failed\n");
+                TEST_ERROR;
+            }
+        }
+
+        if (s->asteps != 0 && step % s->asteps == 0) {
+            if (!verify_dset_attribute(dset_id, which, step)) {
+                fprintf(stderr, "verify_dset_attribute failed\n");
+                TEST_ERROR;
+            }
         }
-        if (s->asteps != 0 && step % s->asteps == 0)
-            verify_dset_attribute(ds, which, step);
     }
 
-    *stepp = step;
+    return true;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Sclose(filespace);
+    }
+    H5E_END_TRY;
 
-out:
-    if (H5Sclose(filespace) < 0)
-        errx(EXIT_FAILURE, "H5Sclose failed");
+    return false;
 }
 
-static void
+static bool
+verify_dsets(state_t s, np_state_t *np, mat_t *mat)
+{
+    unsigned *      nextstep      = NULL;
+    unsigned        finished_step = 0;
+    unsigned        which;
+    exchange_info_t last;
+
+    if (!(nextstep = HDcalloc(s.ndatasets, sizeof(*nextstep)))) {
+        fprintf(stderr, "memory allocation failed\n");
+        TEST_ERROR;
+    }
+
+    do {
+        /* Receive the notice of the writer finishing zoo creation,
+         * including the number of chunks finished and the timestamp
+         */
+        if (s.use_named_pipe && HDread(np->fd_writer_to_reader, &last, sizeof(last)) < 0) {
+            fprintf(stderr, "HDread failed\n");
+            TEST_ERROR;
+        }
+
+        for (which = 0; which < s.ndatasets; which++) {
+            dbgf(1, "step %d which %d\n", nextstep[which], which);
```

Comment:
nextstep is probably used for debugging. But it is never assigned and the debugging information doesn't make sense any more. If it is not used, show remove them.

Reply:
I took it out.